packages/react-components/react-positioning/src/middleware/maxSize.ts

```diff
@@ -1,39 +1,39 @@
-import { detectOverflow } from '@floating-ui/dom';
-import type { Middleware, Side } from '@floating-ui/dom';
+import { size } from '@floating-ui/dom';
+import type { Middleware } from '@floating-ui/dom';
 import type { PositioningOptions } from '../types';
-import { parseFloatingUIPlacement } from '../utils/index';
-
-export function maxSize(autoSize: PositioningOptions['autoSize']): Middleware {
-  return {
-    name: 'maxSize',
-    fn: async middlewareArguments => {
-      const { placement, rects, elements, middlewareData } = middlewareArguments;
-      const basePlacement = parseFloatingUIPlacement(placement).side;
-
-      const overflow = await detectOverflow(middlewareArguments);
-      const { x, y } = middlewareData.shift || { x: 0, y: 0 };
-      const { width, height } = rects.floating;
-
-      const widthProp: Side = basePlacement === 'left' ? 'left' : 'right';
-      const heightProp: Side = basePlacement === 'top' ? 'top' : 'bottom';
+import { getBoundary } from '../utils/getBoundary';
+export interface MaxSizeMiddlewareOptions extends Pick<PositioningOptions, 'overflowBoundary'> {
```

Comment:
This is an improvement, previously our autosize couldn't actually handle alternative overflow boundaries